nyx_client/nyx_client/client.py

```diff
@@ -937,14 +937,19 @@ def delete_circle(self, circle: Circle):
         """
         self.delete_circle_by_name(circle.name)
 
-    def get_connections(self) -> list[Connection]:
+    def get_connections(self, allow_upload: bool | None = None) -> list[Connection]:
         """Lists all connections.
 
+        Args:
+            allow_upload: Filter the connections by allow_upload, defaults to show all.
+
         Returns:
             A list of `Connection` objects
 
         Raises:
             requests.HTTPError: if the API request fails.
         """
-        connections = self._nyx_get(NYX_CONNECTIONS_ENDPOINT)
+        params = dict(allow_upload=allow_upload) if allow_upload is not None else None
```

Comment:
Typically (at least in this repo) we don't instantiate dictionarties using the class syntax but instead do e.g.:
`{"allow_upload": allow_upload}`.
(It also makes a [small perf difference](https://katrin-affolter.ch/de/programming/python-dictionaries) but that's irrelevant here.)